### PR TITLE
[RFC] boot,bootloader: setup the snapd_recovery_kernel grubenv

### PR DESCRIPTION
**Pull request by mvo5**
When making a UC20 system bootable from `snap prepare-image` a
bootenv that is specific to the recovery system needs to get
written so that the bootloader can find the right kernel for
the given recovery system. This is now done as part of the
bootloader.InstallBootConfig() code.

Based on https://github.com/snapcore/snapd/pull/7817

Relevant commit https://github.com/snapcore/snapd/commit/2b051597e2d2ec305293dccf7ffce844530a2660

This probably needs some tweaking but I wanted to propose it still
because it should give some more context to https://github.com/snapcore/snapd/pull/7817 and what we need to do there next to have the full "makeBootable()" story.

### CODE REVIEW

File: bootloader/grub.go [hunk 1]
```diff
@@ -68,13 +69,36 @@ func (g *grub) InstallBootConfig(gadgetDir string, opts *Options) (bool, error)
...
+			return true, err
+		}
+		return true, g.setRecoveryKernel(opts.RecoverySystem, opts.RecoveryKernel)
 	}
 	gadgetFile := filepath.Join(gadgetDir, g.Name()+".conf")
 	systemFile := filepath.Join(g.rootdir, "/boot/grub/grub.cfg")
 	return genericInstallBootConfig(gadgetFile, systemFile)
 }
 
+func (g *grub) setRecoveryKernel(recoverySystem, kernel string) error {
```
Comment (pedronis) [inline bootloader/grub.go]: this seems splitting responsabilites along the wrong lines, it the first time in bootloader that we use/know directly snap_* boot env vars (or about systems), we need to divide this with boot differently. OTOH we don't know how to do this for other bootloaders yet.  Maybe a method only on grub  SetRecoverySystemEnv(recoverySystemDir, values map[string]string), and then using an interface in boot to detect if it is applicable.

Comment (pedronis) [reply bootloader/grub.go]: To be clear, this means not changing InstallBootConfig itself but calling that one separately and explicitly in boot.

**Comment by pedronis** [comment]
@mvo5 also related to this, before I forget, we should remember that the kernel can be in <system-seed>/snaps  or in the snaps directory of the system (if it was unasserted)

**Comment by mvo5** [comment]
Closing and will push a new version based on the feedback from Samuele.

**Pull request closed without merging**